### PR TITLE
1.24 release: finalize change notes for C/C++

### PR DESCRIPTION
**Pull request by felicitymay**
This PR aims to finalize the analysis change notes ready for inclusion with LGTM Enterprise.

The first commit reorders rows in one table.
The [second commit](https://github.com/Semmle/ql/commit/05d0d844bdafb0ea686280576faf45b87cb4db69) makes the following text changes:

- Add a note about the new option for alert suppression comments (as per most other languages).
- Move the "All CWE queries..." row out of the table into a separate paragraph.
- Combine multiple changes to single queries into single rows.
- Reorder the list of queries in another row so they're alphabetic
- Revise the text describing the new interface for allocation and deallocation.

Please check that you're happy with my changes and that I haven't introduced any errors: @Semmle/cpp-analysis

### CODE REVIEW

File: change-notes/1.24/analysis-cpp.md [hunk 1]
```diff
@@ -12,26 +14,29 @@ The following changes in version 1.24 affect C/C++ analysis in all applications.
...
 | Overflow in uncontrolled allocation size (`cpp/uncontrolled-allocation-size`) | Fewer false positive results | The query now produces fewer, more accurate results. Cases where the tainted allocation size is range checked are more reliably excluded. |
 | Overloaded assignment does not return 'this' (`cpp/assignment-does-not-return-this`) | Fewer false positive results | This query no longer reports incorrect results in template classes. |
+| Pointer overflow check (`cpp/pointer-overflow-check`),</br> Possibly wrong buffer size in string copy (`cpp/bad-strncpy-size`),</br> Signed overflow check (`cpp/signed-overflow-check`) | More correct results | A new library is used for determining which expressions have identical value, giving more precise results. There is a performance cost to this, and the LGTM suite will overall run slower than before. |
```
Comment (jbj) [inline change-notes/1.24/analysis-cpp.md:37]: Surely the `</br>` here should be `<br>` or (if you like XML) `<br/>`. It seems like GitHub's Markdown renderer tolerates `</br>`, but we could run into problems with other renderers.

Comment (felicitymay) [reply change-notes/1.24/analysis-cpp.md:37]: Thanks. Apologies for the premature merge 😞. Will fix.

Comment (jbj) [reply change-notes/1.24/analysis-cpp.md:37]: The merge was _not_ premature. It was checked by you and Geoffrey, and two people is enough. I just happened to also read through the changes and noticed that one thing, but I want to be clear that I _don't_ need to sign off on all C++ analysis PRs.

Comment (felicitymay) [reply change-notes/1.24/analysis-cpp.md:37]: Thanks 😄

**Pull request merged**